cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -230,8 +231,17 @@ protected HiveConf getHiveConf() {
       String hadoopAuthToken = System.getenv(UserGroupInformation.HADOOP_TOKEN_FILE_LOCATION);
       if (hadoopAuthToken != null) {
         conf.set("mapreduce.job.credentials.binary", hadoopAuthToken);
+        if ("tez".equals(conf.get("hive.execution.engine"))) {
+          // Add token file location property for tez if engine is tez
+          conf.set("tez.credentials.path", hadoopAuthToken);
+        }
       }
     }
+
+    // Since we use delegation token in HIVE, unset the SPNEGO authentication if it is
+    // enabled. Please see CDAP-3452 for details.
+    conf.unset(HiveConf.ConfVars.HIVE_SERVER2_SPNEGO_KEYTAB.toString());
+    conf.unset(HiveConf.ConfVars.HIVE_SERVER2_SPNEGO_PRINCIPAL.toString());
```

Comment:
Please note that this is separate issue found while testing HIVE on secure cluster. We need to disable SPNEGO authentication if it is enabled in hive-site.